chain-ser/src/packer.rs

```diff
@@ -64,79 +65,89 @@ impl<R: std::io::BufRead> Codec<R> {
     #[inline]
     pub fn get_bytes(&mut self, n: usize) -> Result<Vec<u8>, ReadError> {
         let mut buf = vec![0u8; n];
-        self.0.read_exact(&mut buf)?;
+        self.inner.read_exact(&mut buf)?;
         Ok(buf)
     }
     #[inline]
-    pub fn get_slice(&mut self, slice: &mut [u8]) -> Result<(), ReadError> {
-        self.0.read_exact(slice)?;
+    /// This is a wrapper over the std::io::BufRead::fill_buf() function,
+    /// so be aware of that you need also execute consume() function to move the reader position
```

Comment:
I think we are starting to expose the internal complexity here. Calling `BufRead::consume(n)` inside of this function would be just fine IMO. The original goal of this function is to return a slice bytes and proceed forward.